### PR TITLE
Make compile cleanly with stack --pedantic

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: src/Data/Swagger/Lens.hs [hunk 1]
```diff
@@ -62,17 +62,17 @@ makePrisms ''Referenced
 
 _SwaggerItemsArray :: Review (SwaggerItems 'SwaggerKindSchema) [Referenced Schema]
 _SwaggerItemsArray
-  = prism (\x -> SwaggerItemsArray x) $ \x -> case x of
+  = unto (\x -> SwaggerItemsArray x) {- $ \x -> case x of
```
Comment (phadej) [inline src/Data/Swagger/Lens.hs]: @fizruk I'm wasn't sure, if this and `_SwaggerItemsObject ` are lawful `Prism'`, so changed to `unto`.

File: swagger2.cabal [hunk 2]
```diff
@@ -40,8 +40,8 @@ library
     Data.Swagger.Internal.Utils
     Data.Swagger.Internal.AesonUtils
   build-depends:       base        >=4.7   && <4.10
-                     , base-compat >=0.6.0 && <0.10
-                     , aeson
+                     , base-compat >=0.9.1 && <0.10
+                     , aeson       >=0.11.2.1
```
Comment (phadej) [inline swagger2.cabal]: needed for `ToJSON Day`, only `aeson-0.10` introduced it, but skipping directly to 0.11 should be a problem for any user.

Comment (fizruk) [inline swagger2.cabal]: *Shouldn't* be a problem?

File: src/Data/Swagger/Lens.hs (hunk 1 above)
Comment (fizruk) [inline src/Data/Swagger/Lens.hs]: What made you suspicious? I don't see any violation (not looking very hard).

Comment (phadej) [inline src/Data/Swagger/Lens.hs]: Well, the type is `Review` and not `Prism'`, so I fixed the implementation (term) not the spec (type) :)

File: swagger2.cabal (hunk 2 above)
Comment (phadej) [inline swagger2.cabal]: yes, *shouldn't*.

File: src/Data/Swagger/Lens.hs (hunk 1 above)
Comment (phadej) [inline src/Data/Swagger/Lens.hs]: So should I change type to `Prism'` and try to "prove" the laws in tests?

Comment (fizruk) [inline src/Data/Swagger/Lens.hs]: Ah, I didn't realise I had `Review` type! I wonder why I abandoned `Prism`... perhaps it did not compile with some old GHC. Thanks anyway!

Comment (fizruk) [inline src/Data/Swagger/Lens.hs]: I don't this you should bother, to be honest :)

Comment (phadej) [inline src/Data/Swagger/Lens.hs]: Sorry, cannot parse the previous

Comment (fizruk) [inline src/Data/Swagger/Lens.hs]: Yeah, don't waste your time on this.

**Pull request merged**